### ISSUE TITLE
Remove need for unsafe code

### ISSUE THREAD

**Issue by schungx** [unitycontainer/container#133]
Currently, in order to get the declaring type of the `Parent` in a `ResolverDelegate`, an `unsafe` code block must be used because `Parent` is an `IntPtr`.

Is there any way to avoid having to use an `unsafe` block?

**Comment by schungx** [comment]
Yes, please add it to the structure, or have a way to get this information.  `DeclaringType` is apparently necessary for almost all logging frameworks that categorize based on the type name.

**Comment by ENikS** [comment]
It is available in 4.0.1
